feat: Add confirm quote request dialog component

JedidiahAwuku opened the pull request.

JedidiahAwuku committed:
feat: Add confirm quote request dialog component

github-actions[bot] committed:
Add license header

cawmichae commented on feature-libs/commerce-quotes/assets/translations/en/commerce-quotes.i18n.ts:
```diff
@@ -60,6 +60,16 @@ export const commerceQuotes = {
       quotes: 'Quotes',
     },
+    confirmSubmitDialog: {
+      title: 'Confirm Send Quote 000000123?',
```
Should the quote be hard coded?

Should this be i18n alised?

JedidiahAwuku replied:
Resolved now. Thanks

cawmichae commented on feature-libs/commerce-quotes/assets/translations/en/commerce-quotes.i18n.ts:
```diff
@@ -60,6 +60,16 @@ export const commerceQuotes = {
+
+      warningSubmitNote:
+        'Once a request for quote is submitted it cannot be modified.',
+      confirmSubmitNote: 'Are you sure you want to submit this quote?',
```
i18n?

JedidiahAwuku replied:
Also resolved. Thanks

JedidiahAwuku committed:
refactor: Address code review comments

Remove hardcoded quote Id from I18n file

Change current flow to have dialog show when submit quote is clicked
on the quote details page and not the cart page.
This matches the flow designed on wires.

JedidiahAwuku committed:
fix: failing tests after code refactor

Add new unit test for submit quote confirm dialog

cawmichae commented on feature-libs/commerce-quotes/components/commerce-quotes-actions-by-role/commerce-quotes-actions-by-role.component.ts:
```diff
@@ -4,28 +4,79 @@
+          )
+          .subscribe()
+      );
+    } else {
```
Could put a return statement at the end of the if block, and so remove the else

JedidiahAwuku replied:
Done. Thanks

JedidiahAwuku committed:
refactor: Address comments on code review
